src/java/org/apache/cassandra/transport/messages/BatchMessage.java

```diff
@@ -245,6 +252,12 @@ public BatchMessage(BatchStatement.Type type, List<Object> queryOrIdList, List<L
         }
         catch (Exception e)
         {
+            if(auditLogEnabled)
```

Comment:
Can we have some simple API call form here and then in that static call we do all the business check? I am trying to simplify Messaging/QueryProcessing classes to not add much of business logic.

For example, here we can do something like
```

Audit.java
public static shouldAudit(param1, param2,...)
{
    //add feature to not audit all the queries

    if(ThreadLocalRandom.current().nextDouble() > auditFraction)
    //do not audit this query
        return;

    if (auditLogEnabled) 
    {
        ...
    }
}
```

And then from higher classes like `Message.java`, `ExecuteMessage.java.java` we just simply call
`Audit.shouldAudit(.....)`

Reply:
Alternatively we could delegate this entirely to auditLogManager.log and just have any percentage based choices made in the `IAuditLogger`?

Reply:
Implementers of `IAuditLogger` would have the choice to go with percentage based if they like